### PR TITLE
Add custom unknownHandler

### PR DESCRIPTION
**Pull request by Hocdoc**
As described in #36 this PR adds a new option `unknownHandler` to transform unknown nodes to HAST.

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -83,6 +83,15 @@ The default behaviour is to prefer the last duplicate definition.
 Object mapping [mdast][] [nodes][mdast-node] to functions handling them.
 Take a look at [`lib/handlers/`][handlers] for examples.
 
+###### `options.unknownHandler`
+
+Handler for all unknown nodes.
+
+Default behaviour:
```
Comment (wooorm) [inline readme.md]:
```suggestion
Default behavior:
```

I switched away from en-GB recently

**Comment by Hocdoc** [comment]
Ok, I have changed the test and the README.

(In my project most of the nodes will be non-MDAST-nodes from multiple sources and plugins, so I still think it makes sense to use custom node types.)

File: readme.md [hunk 2]
```diff
@@ -87,7 +87,7 @@ Take a look at [`lib/handlers/`][handlers] for examples.
 
 Handler for all unknown nodes.
 
-Default behaviour:
+Default behavior:
```
Comment (wooorm) [inline readme.md]: ah, thanks for fixing this too!

File: test/handlers-option.js [hunk 3]
```diff
@@ -19,28 +19,33 @@ test('handlers option', function(t) {
     'should override default handler'
   )
 
-  var mdastWithMetaNode = u('paragraph', [
-    u('meta-node', 'meta'),
+  var customMdast = u('paragraph', [
+    u('custom', 'with value'),
+    u('custom', [u('img', {src: 'with-children.png'})]),
```
Comment (wooorm) [inline test/handlers-option.js]: ahhh, sorry for this mistake, I was thinking in hast, not mdast!

```suggestion
    u('custom', [u('image', {url: 'with-children.png'})]),
```

File: test/handlers-option.js [hunk 4]
```diff
@@ -19,28 +19,33 @@ test('handlers option', function(t) {
...
-    to(mdastWithMetaNode, {
+    to(customMdast, {
       unknownHandler: function(n, node) {
         return node
       }
     }),
     u('element', {tagName: 'p', properties: {}}, [
-      u('meta-node', 'meta'),
+      u('custom', 'with value'),
+      u('custom', [u('img', {src: 'with-children.png'})]),
```
Comment (wooorm) [inline test/handlers-option.js]: will this one still work with the above change? 😬

**Comment by Hocdoc** [comment]
No problem. 
I've now changed `img` to `image` for the default-unknown-handler test case.
(In the custom unknown-handler test case I just return the original unknown node).

**Comment by wooorm** [comment]
thanks, released as 8.1.0!

**Pull request merged**